include/deal.II/base/linear_index_iterator.h

```diff
@@ -249,8 +249,15 @@ class LinearIndexIterator
    * Comparison operator. Returns <code>true</code> if both iterators point to
    * the same entry in the same container.
    */
-  bool
-  operator==(const DerivedIterator &) const;
+  template <typename OtherIterator>
+  friend typename std::enable_if<
```

Comment:
What about `std::enable_if_t`? :smile:

Reply:
Oh, sure we can replace that everywhere now if anyone cares.